### PR TITLE
Created API Based Web Tech Jobs Application

### PR DESCRIPTION
**Pull request by Shaheryar0054**
- I choose Jobs API to display tech jobs
- Fetched data from the Jobs API
- Added Home page 
- When the page loads, the web app shows the list of categories that could be filtered out by some parameter, for example by the 
Jobs title.
- Along with the Title of the job, It shows the total number of jobs Available.
- When the user clicks on the job title, the application navigates to the details page.
- Added Details Page
- On the detail page it shows details about the job e.g salary etc.
- Added test cases for the components.

### CODE REVIEW

File: src/components/JobList.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  const frontEndJobsCount = jobs.filter((job) => job.title.toLowerCase().includes('front end')).length;
+
+  const javascriptJobsCount = jobs.filter((job) => job.title.toLowerCase().includes('javascript')).length;
+
+  const fullstackJobsCount = jobs.filter((job) => job.title.toLowerCase().includes('full stack')).length;
+
+  const seniorFullStackJobsCount = jobs.filter((job) => job.title.toLowerCase().includes('senior full stack developer')).length;
+
+  const totalTechJobsCount = jobs.length;
```
Comment (thecodechaser) [inline src/components/JobList.js]: - [ ] You did really a great job on this project but you missed a major requirement. Please add a feature on the home page to filter items(jobs). (You should include a search feature on the page).
image

File: README.md [hunk 2]
```diff
@@ -1,2 +1,140 @@
...
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+- Thanks to the Microverse team for the great curriculum.
+- Thanks to the Code Reviewer(s) for the insightful feedback.
+- A great thanks to My coding partner(s), morning session team, and standup team for their contributions.
+- Hat tip to anyone whose code was used.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
```
Comment (thecodechaser) [inline README.md]: - [ ] Please make sure to give appropriate credit to the author in the README for the original design idea to [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio) as requested in the project requirements.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Shaheryar0054👋, 

### You are doing a marvelous job on the React capstone project and this is one of the best projects I have seen so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- All tests are passed. ✔️
- Highly organized code. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 3]
```diff
@@ -1,2 +1,140 @@
...
+If you like this project please give me stars rating 
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+- Thanks to the Microverse team for the great curriculum.
+- Thanks to the Code Reviewer(s) for the insightful feedback.
+- A great thanks to My coding partner(s), morning session team, and standup team for their contributions.
+- Hat tip to anyone whose code was used.
```
Comment (nahnahdev) [inline README.md]: - [ ] Great job including the search functionality as the previous reviewer mentioned on the homepage 😉 It works as expected. However, I noticed that you failed to give appropriate credit in your README for the original design idea to [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio) as requested in the project requirements and also as mentioned by the previous reviewer. So kindly implement that 💯

File: src/components/JobList.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  const jobStatus = useSelector((state) => state.jobs.status);
+
+  const [searchTerm, setSearchTerm] = useState('');
+
+  useEffect(() => {
+    if (jobStatus === 'idle') {
+      dispatch(fetchJobs());
+    }
+  }, [jobStatus, dispatch]);
+  // eslint-disable-next-line max-len
```
Comment (nahnahdev) [inline src/components/JobList.js]: - [ ] I recommend avoiding the use of `eslint` disable commands in your projects, as these errors and warnings are typically helpful for identifying areas where you can improve your code. By resolving these issues, you can write more clean and readable code that adheres to industry standards, which can make it easier to maintain your codebase over time. :100:

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @Shaheryar0054 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which haven't been addressed yet.


 ## Highlights
- No linter errors ✔️ 
- Search functionality added ✔️ 
- App works as expected ✔️ 
- All tests passing ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @NahnahAJ in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @Shaheryar0054 ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

## Optional suggestions

- [ ] You may always write your commit messages in an imperative tense something like `Add Home Button` instead of `Added Home Button`.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**